dimscord/dispatch.nim

```diff
@@ -20,6 +21,30 @@ proc checkIfAwaiting(client: DiscordClient, event: static[DispatchEvent], data:
       # Remove the handler if it gets completed
       handlers[].del(i)
 
+macro checkAndCall(shard: Shard, event: static[DispatchEvent], args: varargs[untyped]) =
+  ## Checks if any handlers are waiting for an event and then calls the users handler
+  # Convert the passed in values into a list
+  let params = collect:
+    for arg in args:
+      arg
+
+  let
+    tupleData = nnkTupleConstr.newTree(params)
+    eventName = ident toLowerAscii($event)
+    client = shard.newDotExpr(ident"client")
+    # Generate version of call without shard (Not every event takes the shard)
+    call = client.newDotExpr(ident"events")
+                 .newDotExpr(eventName)
+                 .newCall(params)
+    # Generate version with shard
+    callWithShard = call.dup(insert(1, shard))
+  result = quote do:
+    `client`.checkIfAwaiting(`event`, `tupleData`)
+    when compiles(`callWithShard`):
+      await `callWithShard`
+    else:
+      await `call`
```

Comment:
it should be `asyncCheck`, since we don't what events to affect the background process of the gateway/restapi stuffs.